### PR TITLE
Add registers.site to context when rendering template

### LINKED ISSUE #101: Jekyll 4.2.0 breaks this plugin
I just upgraded to [Jekyll 4.2.0](https://jekyllrb.com/docs/history/#v4-2-0) (which was released recently) and noticed that I get an error upon attempting to run my project.

Here is the stacktrace:

```
Configuration file: [...]/_config.yml
Configuration file: [...]/_config_dev.yml
            Source: [...]
       Destination: [...]/build_dev
 Incremental build: disabled. Enable with --incremental
      Generating...
Creating output directory [...]/build_dev/assets/media/r
Generating [...]/build_dev/assets/media/r/imagename.jpg
  Liquid Exception: undefined method `filter_cache' for nil:NilClass in [...]/_posts/postname.md
bundler: failed to load command: jekyll ([...]/.rbenv/versions/2.6.3/bin/jekyll)
NoMethodError: undefined method `filter_cache' for nil:NilClass
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/filters.rb:425:in `item_property'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/filters.rb:385:in `block in sort_input'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/filters.rb:385:in `map'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/filters.rb:385:in `sort_input'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/filters.rb:320:in `sort'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/strainer.rb:56:in `invoke'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/context.rb:86:in `invoke'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/variable.rb:84:in `block in render'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/variable.rb:82:in `each'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/variable.rb:82:in `inject'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/variable.rb:82:in `render'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/tags/assign.rb:26:in `render'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/block_body.rb:103:in `render_node_to_output'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/block_body.rb:91:in `render'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:208:in `block in render'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:242:in `with_profiling'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:207:in `render'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:220:in `render!'
  [...]lib/jekyll-responsive-image/renderer.rb:28:in `render_responsive_image'
  [...]lib/jekyll-responsive-image/tag.rb:16:in `render'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/block_body.rb:103:in `render_node_to_output'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/block_body.rb:91:in `render'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:208:in `block in render'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:242:in `with_profiling'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:207:in `render'
  [...]lib/ruby/gems/2.6.0/gems/liquid-4.0.3/lib/liquid/template.rb:220:in `render!'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:39:in `block (3 levels) in render!'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:59:in `measure_counts'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:38:in `block (2 levels) in render!'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:63:in `measure_bytes'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:37:in `block in render!'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:70:in `measure_time'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/liquid_renderer/file.rb:36:in `render!'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:131:in `render_liquid'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:80:in `render_document'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/renderer.rb:63:in `run'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:547:in `render_regenerated'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:532:in `block (2 levels) in render_docs'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:531:in `each'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:531:in `block in render_docs'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:530:in `each_value'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:530:in `render_docs'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:210:in `render'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/site.rb:80:in `process'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/command.rb:28:in `process_site'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/commands/build.rb:65:in `build'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/commands/build.rb:36:in `process'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/command.rb:91:in `block in process_with_graceful_fail'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/command.rb:91:in `each'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/command.rb:91:in `process_with_graceful_fail'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/lib/jekyll/commands/build.rb:18:in `block (2 levels) in init_with_program'
  [...]lib/ruby/gems/2.6.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `block in execute'
  [...]lib/ruby/gems/2.6.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `each'
  [...]lib/ruby/gems/2.6.0/gems/mercenary-0.4.0/lib/mercenary/command.rb:221:in `execute'
  [...]lib/ruby/gems/2.6.0/gems/mercenary-0.4.0/lib/mercenary/program.rb:44:in `go'
  [...]lib/ruby/gems/2.6.0/gems/mercenary-0.4.0/lib/mercenary.rb:21:in `program'
  [...]lib/ruby/gems/2.6.0/gems/jekyll-4.2.0/exe/jekyll:15:in `<top (required)>'
  [...]bin/jekyll:23:in `load'
  [...]bin/jekyll:23:in `<top (required)>'
```

If I have some time I might file a PR as I have some other modifications in a fork I'd like to have considered upstream.

### PR DESCRIPTION
**Pull request by salomvary**
Fixes #101.

### CODE REVIEW

**Comment by Lominean** [comment]
No bot, go away!

File: features/test-site/_includes/responsive-image.html [hunk 1]
```diff
@@ -1 +1,3 @@
+{% assign largest = resized | sort: 'width' | last %}
```
Comment (wildlyinaccurate) [inline features/test-site/_includes/responsive-image.html:1]: Is this change intentional?

Comment (salomvary) [reply features/test-site/_includes/responsive-image.html:1]: @wildlyinaccurate This line triggers the bug described in #103. I should have left some comment explaining the purpose, sorry about that.

**Comment by brandonb927** [comment]
This is not stale

**Comment by trombik** [comment]
confirmed the PR fixed the issue in one of my repositories.

**Comment by wildlyinaccurate** [comment]
Thanks @salomvary for this fix! Thanks everyone else who commented and tested. Really sorry this took so long to merge. I'll cut a new gem version now.
